ops/jenkins/global-pipeline-libraries/vars/awsSsm.groovy

```diff
@@ -25,3 +25,13 @@ String putParameter(Map args = [:]) {
     output = sh(returnStdout: true, script: "aws ssm put-parameter --name ${name} --value '${value}' ${includeType} --region ${awsRegion} ${overwrite}").trim()
     return output
 }
+
+// Adds or overwrites one or more tags for the specified resource
+String tagResource(Map args = [:]) {
+    type = args.resourceType ?: 'Parameter'
+    id = args.resourceId
+    tags = args.resourceTags
+
+    output = sh(returnStdout: true, script: "aws ssm add-tags-to-resource --resource-type ${type} --resource-id ${id} --tags ${tags}").trim()
+    return output
+}
```

Comment:
I wonder if this shouldn't be abstracted away. Any objections to introducing `parameterTags` to `putParameter()` and making an implicit call to `tagResource()` when the value of `parameterTags` is a non-null value? Reduced to a single call to `putParameter()`, usage might look like the following in the case of the `bfd-data-fda` Jenkinsfile:

```groovy
awsSsm.putParameter(
    parameterName: "/bfd/mgmt/versioning/nonsensitive/bfd-data-fda-sha256",
    parameterValue: "${fdaDrugCodeSha256}",
    parameterType: "String",
    parameterTags: "Key=Source,Value=${JOB_NAME} Key=Environment,Value=mgmt Key=stack,Value=mgmt Key=Terraform,Value=False Key=application,Value=bfd Key=business,Value=oeda",   
    shouldOverwrite: true
)
```

Reply:
So I made a change to include everything needed to tag the parameter in the putParameter() function. I think if we are creating a parameter, we should also default tag it if tags are not provided. I tried to call this as a separate function but still need to call that send command to tag anything anyway. Why not just add the tag CLI command by default in this function? Let me know what you think.

Reply:
> So I made a change to include everything needed to tag the parameter in the putParameter() function. I think if we are creating a parameter, we should also default tag it if tags are not provided. I tried to call this as a separate function but still need to call that send command to tag anything anyway. Why not just add the tag CLI command by default in this function? Let me know what you think.

That makes a lot of sense. I can't think of a situation where we wouldn't want the additional context a tag might provide. It wouldn't have occurred to me to use the `$JOB_NAME` in the global pipeline library, but _of course_ it's available as [a default environment variable](https://jenkins-east.cloud.cms.gov/bfd/env-vars.html)–nice!

I don't think there's all that much to reason about here and it looks like it ought to work. While it could be more idiomatic groovy, like [terraform.goovy's](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/ops/jenkins/global-pipeline-libraries/vars/terraform.groovy#L15-L19) handling of terraform variable key/value pairs, I don't think there's great reason to _not_ accept this change set. Let's go!